src/aarch64/Gstep.c

```diff
@@ -119,6 +119,8 @@ aarch64_handle_signal_frame (unw_cursor_t *cursor)
   c->dwarf.loc[UNW_AARCH64_SP]  = DWARF_LOC (sc_addr + LINUX_SC_SP_OFF, 0);
   c->dwarf.loc[UNW_AARCH64_PC]  = DWARF_LOC (sc_addr + LINUX_SC_PC_OFF, 0);
   c->dwarf.loc[UNW_AARCH64_PSTATE]  = DWARF_LOC (sc_addr + LINUX_SC_PSTATE_OFF, 0);
+  for (reg = 0; reg < 32; ++reg)
+    c->dwarf.loc[UNW_AARCH64_V0 + reg] = DWARF_LOC (&c->dwarf, LINUX_SC_V0_OFF + reg * 0x10);
```

Comment:
Do you have a test or an example that shows this working? I have a suspicion that it might not because of my previos comment and the default dwarf_get() function is for 64 bit values and these V registers are 128 bit.

Reply:
This seems to go through `dwarf_getfp`, which is 128-bit values (unw_fpreg_t is a long double). This PR is segfaulting in that method though on a few tests (Ltest-resume-sig), so I see this PR is very much not working correctly though.

Reply:
That test now passes for me, and it attempts to do a bit of FP state manipulation (but not these vectors). I am not sure of a particularly direct or easy way to check that it is getting it correct.

Reply:
Thanks I see that the latest change looks better with the new macro. I just worry that without a before and after of something working vs not working that we might be adding something that is subtly incorrect that will just need to be fixed up later.

One other issue is that the floating point registers are only saved in the signal context if system_supports_fpsimd() is true in the kernel. To be honest I'm not sure if that's an issue in practice as these would only be accessed in the case where it was enabled anyway.

I actually have a very similar change that I'm just about to post in the next couple of days that adds support for SVE unwinding where I check for the existence of the SVE section before assigning the locations. Perhaps you could use the same search function to search for the FP section magic number, although as I said maybe in practice we can be certain that it's always there.

Reply:
Yeah, that is fair, and I don't actually have a use-case for this, just an observation. Do you want to just take this over then, since it is likely just a sub-case of your vfp work?

Looking at `src/aarch64/Gresume.c`, we can actually be reasonably sure we don't need to care about these in most typical cases for `unw_resume` at least. In particular, normally for `unw_resume` it only restores the non-scratch register state. While for signals, it will implicitly inherit all unlisted state from signal. So as long as none of these registers are considered callee-saved, there shouldn't be an issue for code following the normal C ABI.

There is a slight hiccup with that claim, which is that `unw_setcontext` is currently a macro (rather than a proper assembly function), and therefore does not correctly force it to use the platform ABI. But that is tangential to this change.

Reply:
It's been a while because I was waiting for the kernel side changes to be merged, but here's the change I was talking about: https://github.com/libunwind/libunwind/pull/423/commits/01e093b09d521d1432c40972e5533c19f69e18f3#diff-d8a7e9d46e2dc8da2f5bef4f1ba9ae27cfdecc24e8035d294b4f2783c26c523fR84

See "switch (magic)", you could add the FP magic number in that switch and then add the dwarf locs if it's found.